Snowflake.Data.Tests/UnitTests/Tools/UnixOperationsTest.cs

```diff
@@ -45,42 +47,79 @@ public void TestDetectGroupOrOthersWritablePermissions(
                 Assert.Ignore("skip test on Windows");
             }
-            
+
             // arrange
             var filePath = CreateConfigTempFile(s_workingDirectory, "random text");
             var readWriteUserPermissions = FilePermissions.S_IRUSR | FilePermissions.S_IWUSR;
             var filePermissions = readWriteUserPermissions | groupOrOthersWritablePermissions | groupNotWritablePermissions | otherNotWritablePermissions;
             Syscall.chmod(filePath, filePermissions);
-            
+
             // act
             var result = s_unixOperations.CheckFileHasAnyOfPermissions(filePath, FileAccessPermissions.GroupWrite | FileAccessPermissions.OtherWrite);
-            
+
             // assert
             Assert.IsTrue(result);
         }
 
         [Test]
         public void TestDetectGroupOrOthersNotWritablePermissions(
-            [ValueSource(nameof(UserPermissions))] FilePermissions userPermissions, 
+            [ValueSource(nameof(UserPermissions))] FilePermissions userPermissions,
             [ValueSource(nameof(GroupNotWritablePermissions))] FilePermissions groupNotWritablePermissions,
             [ValueSource(nameof(OtherNotWritablePermissions))] FilePermissions otherNotWritablePermissions)
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
                 Assert.Ignore("skip test on Windows");
             }
-            
+
             var filePath = CreateConfigTempFile(s_workingDirectory, "random text");
             var filePermissions = userPermissions | groupNotWritablePermissions | otherNotWritablePermissions;
             Syscall.chmod(filePath, filePermissions);
-            
+
             // act
             var result = s_unixOperations.CheckFileHasAnyOfPermissions(filePath, FileAccessPermissions.GroupWrite | FileAccessPermissions.OtherWrite);
-            
+
             // assert
             Assert.IsFalse(result);
         }
 
+        [Test]
+        public void TestReadAllTextCheckingPermissions()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Assert.Ignore("skip test on Windows");
+            }
+            var content = "random text";
+            var filePath = CreateConfigTempFile(s_workingDirectory, content);
+            var filePermissions = FileAccessPermissions.UserReadWriteExecute;
+            Syscall.chmod(filePath, (FilePermissions)filePermissions);
+
+            // act
+            var result = s_unixOperations.ReadAllText(filePath, TomlConnectionBuilder.GetFileValidations());
+
+            // assert
+            Assert.AreEqual(content, result);
+        }
+
+        [Test]
+        public void TestShouldThrowExceptionIfOtherPermissionsIsSetWhenReadAllText([ValueSource(nameof(UserReadWritePermissions))] FilePermissions userPermissions,
+            [ValueSource(nameof(GroupOrOthersWritablePermissions))] FilePermissions groupOrOthersWritablePermissions,
```

Comment:
shouldn't it just be any group permission?

Reply:
Fixed, changed parameters order for readability